spec/requests/v1/pacbio/runs_spec.rb

```diff
@@ -2,27 +2,51 @@
 
 require 'rails_helper'
 
+RSpec::Matchers.define :array_of_aliquots_with_length do |expected_length|
+  match do |actual|
+    actual.is_a?(Array) &&
+      actual.all? { |item| item.is_a?(Aliquot) } &&
+      actual.length == expected_length
+  end
+
+  failure_message do |actual|
+    "expected an array of Aliquot objects with length #{expected_length}, but got #{actual.inspect}"
+  end
+end
+
 RSpec.describe 'RunsController' do
   # Create default and non-default smrt link versions for runs
   let!(:version10) { create(:pacbio_smrt_link_version, name: 'v10') }
   let!(:version11) { create(:pacbio_smrt_link_version, name: 'v11') }
   let!(:version12) { create(:pacbio_smrt_link_version, name: 'v12_revio') }
   let!(:version13) { create(:pacbio_smrt_link_version, name: 'v13_revio', default: true) }
 
-  shared_examples 'publish_messages_on_create' do
+  shared_examples 'publish_messages_on_create' do |aliquot_size|
     it 'publishes a message' do
       expect(Messages).to receive(:publish).with(instance_of(Pacbio::Run), having_attributes(pipeline: 'pacbio'))
       post v1_pacbio_runs_path, params: body, headers: json_api_headers
       expect(response).to have_http_status(:success), response.body
     end
+
+    it 'publishes volume tracking message for each used aliquot' do
+      expect(Emq::Publisher).to receive(:publish).with(array_of_aliquots_with_length(aliquot_size), instance_of(Pipelines::Configuration::Item), 'volume_tracking')
```

Comment:
I think we should just be testing that the message is sent correctly here. The contents of the message should be tested by writing tests for the all_library_aliquots function.

Reply:
Yeah similar to the test above it just checks that a message is received with some data.